### PR TITLE
Running CI with latest develop for "Rebasing put frame return error codes support start stop"

### PR DESCRIPTION
**Pull request by stefankiesz**
https://github.com/awslabs/amazon-kinesis-video-streams-producer-sdk-cpp/pull/1078

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by hassanctech — APPROVED** [review]
lgtm

File: src/KinesisVideoStream.cpp [hunk 1]
```diff
@@ -19,17 +19,17 @@ KinesisVideoStream::KinesisVideoStream(const KinesisVideoProducer& kinesis_video
     }
 }
 
-bool KinesisVideoStream::putFrame(KinesisVideoFrame& frame) const {
-    if (debug_dump_frame_info_) {
+STATUS KinesisVideoStream::putFrame(KinesisVideoFrame& frame) const {
+ //   if (debug_dump_frame_info_) {
```
Comment (stefankiesz) [inline src/KinesisVideoStream.cpp:23]: Uncomment

File: src/gstreamer/gstkvssink.cpp [hunk 2]
```diff
@@ -1374,6 +1378,11 @@ gst_kvs_sink_handle_buffer (GstCollectPads * pads,
         gst_buffer_unref (buf);
     }
 
+    if (STATUS_FAILED(put_frame_status)) {
+        GST_ELEMENT_WARNING (kvssink, STREAM, FAILED, (NULL),
```
Comment (stefankiesz) [inline src/gstreamer/gstkvssink.cpp:1382]: [Copied from original PR]

Can we consider changing this to:

<pre style="color:blue;">
GST_ELEMENT_WARNING (kvssink, <b>RESOURCE</b>, <b>GST_RESOURCE_ERROR_WRITE,</b> (NULL),
</pre>

Based on these descriptions in GStreamer documentation:
> **Stream** errors are for anything related to the stream being processed: format errors, media type errors, ... They're typically used by decoders, demuxers, converters, ...
[here](https://gstreamer.freedesktop.org/documentation/gstreamer/gsterror.html?gi-language=c#GstStreamError:~:text=Stream%20errors%20are%20for%20anything%20related%20to%20the%20stream%20being%20processed%3A%20format%20errors%2C%20media%20type%20errors%2C%20...%20They%27re%20typically%20used%20by%20decoders%2C%20demuxers%2C%20converters%2C%20...)

> **Resource** errors are for any resource used by an element: memory, files, network connections, process space, ... They're typically used by source and sink elements.
[here](https://gstreamer.freedesktop.org/documentation/gstreamer/gsterror.html?gi-language=c#GstResourceError:~:text=Resource%20errors%20are%20for%20any%20resource%20used%20by%20an%20element%3A%20memory%2C%20files%2C%20network%20connections%2C%20process%20space%2C%20...%20They%27re%20typically%20used%20by%20source%20and%20sink%20elements.)

**Review by hassanctech — APPROVED** [review]
LGTM

Looks like we never had tests when putFrame failed.  Let's merge this and have a follow up PR where we have some tests that test specific failures for putFrame as well.

**Review by niyatim23 — APPROVED** [review]

**Pull request merged**